src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/ResettlementAssessmentService.kt

```diff
@@ -489,13 +492,25 @@ class ResettlementAssessmentService(
     )
   }
 
-  fun processProfileTags(resettlementAssessmentEntity: ResettlementAssessmentEntity): List<TagAndQuestionMapping> {
-    val tagList = mutableListOf<TagAndQuestionMapping>()
+  fun getProfileTag(questionId: String, answer: Answer<*>, pathway: Pathway): String? {
+    TagAndQuestionMapping.entries.forEach {
+      if ((questionId == it.questionId) &&
+        (answer.answer.toString().contains(it.optionId)) &&
+        (pathway.name == it.pathway.name)
+      ) {
+        return it.name
+      }
+    }
+    return null
+  }
+
+  fun processProfileTags(resettlementAssessmentEntity: ResettlementAssessmentEntity, pathway: Pathway): List<String> {
+    val tagList = mutableListOf<String>()
     resettlementAssessmentEntity.assessment.assessment.forEach {
-      if (it.questionId == TagAndQuestionMapping.NO_FIXED_ABODE.questionOptionId.toString()) {
-        tagList.add(TagAndQuestionMapping.NO_FIXED_ABODE)
-      } else if (it.questionId == TagAndQuestionMapping.HOME_ADAPTION_POST_RELEASE.questionOptionId.toString()) {
-        tagList.add(TagAndQuestionMapping.HOME_ADAPTION_POST_RELEASE)
+      log.info("Answer value for question id ${it.questionId} is ${it.answer.answer}")
```

Comment:
I dont think we need to log each question and answer, some of the answers are considered sensitive and so we shouldnt put them in the logs

Reply:
Added the log for local debug, forgot to remove it. it is done now!